cocos/ui/UIPageView.h

```diff
@@ -111,81 +107,108 @@ class CC_GUI_DLL PageView : public Layout
+    CC_DEPRECATED_ATTRIBUTE void addWidgetToPage(Widget* widget, ssize_t pageIdx, bool forceCreate);
     
     /**
      * Insert a page into the end of PageView.
      *
      * @param page Page to be inserted.
+	 *
+	 * Since v3.9, this is deprecated. Use `ListView::pushBackCustomItem(Widget* item)` instead.
      */
-    void addPage(Layout* page);
+    CC_DEPRECATED_ATTRIBUTE void addPage(Layout* page);
     
     /**
      * Insert a page into PageView at a given index.
      *
      * @param page  Page to be inserted.
      * @param idx   A given index.
+	 *
+	 * Since v3.9, this is deprecated. Use `ListView::insertCustomItem(Widget* item, ssize_t index)` instead.
      */
-    void insertPage(Layout* page, int idx);
+    CC_DEPRECATED_ATTRIBUTE void insertPage(Layout* page, int idx);
     
     /**
      * Remove a page of PageView.
      *
      * @param page  Page to be removed.
-     */
-    void removePage(Layout* page);
+	 *
+	 * Since v3.9, this is deprecated. Use `ListView::removeItem(getIndex(item))` instead.
+	 */
+	CC_DEPRECATED_ATTRIBUTE void removePage(Layout* page);
 
     /**
      * Remove a page at a given index of PageView.
      *
      * @param index  A given index.
-     */
-    void removePageAtIndex(ssize_t index);
+	 *
+	 * Since v3.9, this is deprecated. Use `ListView::removeItem(ssize_t index)` instead.
+	 */
+	CC_DEPRECATED_ATTRIBUTE void removePageAtIndex(ssize_t index);
 
     /**
-     * Changes scroll direction of PageView
-     *
-     * @see `Direction`
-     * @param direction Scroll direction enum.
-     * @since v3.8
-     */
-    void setDirection(Direction direction);
-
-    /**
-     * Query scroll direction of PageView.
-     *
-     * @see `Direction`
-     * @since v3.8
-     * @return PageView scroll direction.
-     */
-    Direction getDirection()const;
-    
-    /**
      * @brief Remove all pages of the PageView.
-     */
-    void removeAllPages();
+	 *
+	 * Since v3.9, this is deprecated. Use `ListView::removeAllItems()` instead.
+	 */
+	CC_DEPRECATED_ATTRIBUTE void removeAllPages();
     
     /**
      * Scroll to a page with a given index.
      *
      * @param idx   A given index in the PageView. Index start from 0 to pageCount -1.
+	 *
+	 * Since v3.9, this is deprecated. Use `ListView::scrollToItem(ssize_t itemIndex, Vec2::ANCHOR_MIDDLE, Vec2::ANCHOR_MIDDLE)` instead.
+	 */
+	CC_DEPRECATED_ATTRIBUTE void scrollToPage(ssize_t idx);
+
+    /**
+     * Scroll to a page with a given index.
+     *
+     * @param idx   A given index in the PageView. Index start from 0 to pageCount -1.
      */
-    void scrollToPage(ssize_t idx);
+    void scrollToItem(ssize_t itemIndex);
 
+    /**
+     * Gets current displayed page index.
+     * @return current page index.
+     *
+     * Since v3.9, this is deprecated. Use `getCurrentPageIndex()` instead.
+     */
+    CC_DEPRECATED_ATTRIBUTE ssize_t getCurPageIndex() const;
```

Comment:
If we mark these method as deprecated, then they won't be exported in JS and Lua binding. So we have to manually write some wrappers in Deprecated.js or Deprecated.lua file.